network-libp2p/src/discovery/behaviour.rs

```diff
@@ -288,10 +299,8 @@ impl NetworkBehaviour for Behaviour {
                     .push_back(ToSwarm::NewExternalAddrCandidate(observed_address));
             }
             HandlerOutEvent::Update => self.events.push_back(ToSwarm::GenerateEvent(Event::Update)),
-            HandlerOutEvent::Error(_) => self.events.push_back(ToSwarm::CloseConnection {
-                peer_id,
-                connection: CloseConnection::All,
-            }),
+            // Errors must not result in a closed connection as light clients are unable to verify ValidatorRecord.
+            HandlerOutEvent::Error(error) => log::trace!(?error, "Received invalid contact"),
```

Comment:
Mmm, but we panic if a light client tries to verify a validator record :thinking: